staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/handler.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregated
+
+import (
+	"net/http"
+	"sync"
+
+	apidiscoveryv2beta1 "k8s.io/api/apidiscovery/v2beta1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"github.com/emicklei/go-restful/v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/handlers/negotiation"
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
+)
+
+const DiscoveryEndpointRoot = "/discovery"
+
+// This handler serves the /discovery/<version> endpoint for a given list of
+// api resources indexed by their group version.
+type ResourceManager interface {
```

Comment:
Couldn't we just move this into discovery/root.go as a separate method ServeHTTPAggregated?  Do we really need two different types implementing the same interface?

Reply:
These are different interfaces since one takes groups and the other includes groups along with their resources. I think the separation makes it more clear that we have two separate handlers for the unaggregated vs aggregated.